src/Tooltip/Tooltip.tsx

```diff
@@ -8,37 +12,92 @@ export interface TooltipMessageGroupItem {
     message: string;
 }
 
-interface TooltipState {
-    toggle: boolean;
-}
-
+export type TooltipTrigger = "hover" | "click" | "focus";
 export type TooltipTheme = "default" | "light" | "primary" | "warning" | "success" | "danger" | "purple";
 export type TooltipPosition = "top" | "bottom" | "left" | "right" | "top-right" | "top-left" | "bottom-right" | "bottom-left" | "left-top" | "left-bottom" | "right-top" | "right-bottom";
 
 export interface TooltipProps {
     className?: string;
     customSvg?: React.ReactNode;
     id?: string;
+    ref?: React.LegacyRef<Tooltip>;
+    /** @deprecated use content instead */
     message?: string;
+    /** @deprecated use content instead */
     messageGroup?: Array<TooltipMessageGroupItem>;
-    onClick?: (event?: React.MouseEvent<HTMLDivElement>) => void;
+    /** @deprecated use onVisibleChange instead */
+    onClick?: (event?: React.MouseEvent<HTMLDivElement> | React.FocusEvent<HTMLElement>) => void;
     position?: TooltipPosition;
     theme?: TooltipTheme;
+    /** @deprecated use content instead */
     title?: string;
+    /** @deprecated use trigger instead */
     triggerOnHover?: boolean;
+    /** @deprecated */
     width?: number;
+    content?: string | React.ReactNode;
+    trigger?: TooltipTrigger;
+    disableAutoPosition?: boolean;
+    onVisibleChange?: (event: React.MouseEvent<HTMLDivElement> | React.FocusEvent<HTMLElement>, visible: boolean) => void;
+    children?: React.ReactNode;
```

Comment:
children here is a react reserve word and would be confusing to users. can you use a name that tallies with the variable intent ?

Reply:
removing the children attribute as the children here is the actual children of the component. I added the attribute to the props interface as it shows error when the component is written as function component